packages/next/shared/lib/router/utils/routing-items.ts

```diff
@@ -0,0 +1,40 @@
+import { getMiddlewareRegex } from './get-middleware-regex'
+import { getRouteMatcher } from './route-matcher'
+import { getRouteRegex } from './route-regex'
+import { getSortedRoutes } from './sorted-routes'
+
+const MIDDLEWARE_SUFFIX = '/_middleware'
+
+export interface RoutingItem {
+  page: string
+  match: ReturnType<typeof getRouteMatcher>
+  ssr?: boolean
+  isMiddleware?: boolean
+}
+
+export function* getRoutingItems(
```

Comment:
Can we avoid using a generator here? This needs to be transpiled for older browser currently which adds half a KB gzipped to the bundle x-ref: https://github.com/vercel/next.js/pull/32601#issuecomment-1029384673

Reply:
@ijjk changed to return Iterator instead.

Reply:
Iterators require the same polyfilling as generators 😅 can we just return an array here?

Reply:
Iterators require the same polyfilling as generators unfortunately 😅 can we just return an array here? Note: we can see the size increase here https://github.com/vercel/next.js/pull/32601#issuecomment-1033886596 which is gzipped so it's quite a bit non-gzipped

Reply:
This computation will be relatively high cost but I'm not sure if we can cache the result (maybe not since pages and middleware can change with hot-reloading). I will change it to an array for now.